tidalapi/session.py

```diff
@@ -644,7 +646,7 @@ def load_session_from_file(self, session_file: Path):
 
         return self.load_oauth_session(**args)
 
-    def _login_with_link(self) -> Tuple[LinkLogin, concurrent.futures.Future[Any]]:
+    def _login_with_link(self) -> JsonObj:
```

Comment:
This method should not be a private method if 3rd party libs plan to access it directly. I also suggest adding doxygen for the same reason as above.

It should also be renamed, since its purpose will have changed significantly. Perhaps something like `oauth_request_device_auth()`

Reply:
You are totally right. Excuse me this sloppy PR. I will do a proper refactoring and re-submit this PR.